src/blocks/transformers/temporaryInfo/temporaryPanelProtocol.ts

```diff
@@ -19,14 +19,32 @@ import { type UUID } from "@/core";
 import pDefer, { type DeferredPromise } from "p-defer";
 import { expectContext } from "@/utils/expectContext";
 import { CancelError } from "@/errors/businessErrors";
+import { type TemporaryPanelEntry } from "@/sidebar/types";
 
-const panels = new Map<UUID, DeferredPromise<void>>();
+type RegisteredPanel = {
+  entry: TemporaryPanelEntry;
+  registration: DeferredPromise<void>;
+};
+
+const panels = new Map<UUID, RegisteredPanel>();
+
+export async function getPanelDefinition(
+  nonce: UUID
+): Promise<TemporaryPanelEntry> {
+  expectContext("contentScript");
+
+  return panels.get(nonce).entry;
```

Comment:
Could this be undefined? We have `?.` on new line 74
```suggestion
  return panels.get(nonce)?.entry;
```

Reply:
I clarified and added a test. The one for form throws and error, and we want this one too. The caller should never be calling with an undefined form